src/modules/modifierOSites/process.cpp

```diff
@@ -0,0 +1,114 @@
+        return ExecutionResult::Failed;
+    }
+    auto &processingData = moduleContext.dissolve().processingModuleData();
+
+    // Select all potential bridging oxygen sites - we will determine which actually are
+    // involved in NF-BO-NF interactions once we have the available NF sites
+    SiteSelector allOxygenSites(targetConfiguration_, bridgingOxygenSpeciesSites_);
+
+    // Select all NF centres
+    SiteSelector NF(targetConfiguration_, networkFormerSpeciesSites_);
+
+    // Select all modifier centres
+    SiteSelector modifier(targetConfiguration_, modifierSpeciesSites_);
+
+    // Filter the oxygen sites into those surrounded by up to two NF sites
+    SiteFilter ofilter(targetConfiguration_, allOxygenSites.sites());
+    auto &&[filteredOSites, neighbourMap] = ofilter.filterBySiteProximity(NF.sites(), distanceRange_, 0, 2);
+
+    SiteFilter mfilter(targetConfiguration_, modifier.sites());
+    auto &&[filteredMSites, mNeighbourMapO] =
+        mfilter.filterBySiteProximity(allOxygenSites.sites(), modifierDistanceRange_, 0, 99);
+
+    // Retrieve storage for the Mofifier to Oxygen Type Sites histogram
+    auto [oxygenSitesHistogram, status] =
+        processingData.realiseIf<IntegerHistogram1D>("OTypeSitesHistogram", name(), GenericItem::InRestartFileFlag);
+    if (status == GenericItem::ItemStatus::Created)
+        oxygenSitesHistogram.initialise();
+
+    // Retrieve storage for the Total O Sites histogram
+    auto [modifierHistogram, status2] =
+        processingData.realiseIf<IntegerHistogram1D>("TotalOHistogram", name(), GenericItem::InRestartFileFlag);
+    if (status2 == GenericItem::ItemStatus::Created)
+        modifierHistogram.initialise();
+
+    // Clear the temporary bins
+    modifierHistogram.zeroBins();
+    oxygenSitesHistogram.zeroBins();
+
+    // The returned 'neighbourMap' maps BO sites to nearby NF sites *only if* there were exactly two NF sites within range.
+    // So, we can use this to determine the Q numbers for each NF by counting the number of times a NF site appears in the map.
+    std::map<const Site *, int> qSpecies;
+    std::map<int, int> oxygenSites;
+    for (const auto &[siteM, nearO] : mNeighbourMapO)
+    {
+        modifierHistogram.bin(nearO.size());
+        for (auto &&[oSite, index] : nearO)
+        {
+            oxygenSitesHistogram.bin(neighbourMap[oSite].size());
+        }
+    }
+
+    // Accumulate histogram averages
+    oxygenSitesHistogram.accumulate();
+    modifierHistogram.accumulate();
+
+    // Averaged values for OSites
+    Data1D accumulatedData = oxygenSitesHistogram.accumulatedData();
+    auto sum = Integrator::absSum(oxygenSitesHistogram.data());
+    accumulatedData /= sum;
+
+    // Average values for total O sites
+    Data1D accumulatedModifierData = modifierHistogram.accumulatedData();
+    auto totalOSites = Integrator::absSum(modifierHistogram.data());
+    accumulatedModifierData /= totalOSites;
+
+    // Create the display data
```

Comment:
```suggestion
    // Store the normalised data
```
But see comment above!